extension-workspace-reference.md

```diff
@@ -0,0 +1,121 @@
+# workspace/xreference extension to LSP
+
+The `workspace/xreference` extension to the Language Server Protocol (LSP) enables a language server to export all of the references to dependencies that code inside of the workspace makes.
+
+Use case: clients of a language server can invoke `workspace/xreference` in order to find external references to dependencies. This information can then be stored inside of a database, which allows the caller to create a 'global mapping' of symbols in dependencies to the workspace they are used in (e.g. to see "how do other people use this symbol?").
+
+### Initialization
+
+`ServerCapabilities` may contain a new field to indicate server-side support for this extension:
+
+```typescript
+interface ServerCapabilities {
+  /* ... all fields from the base ServerCapabilities ... */
+
+  /**
+   * The server provides workspace reference exporting support.
+   */
+  xworkspaceReferenceProvider?: boolean;
+}
+```
+
+#### Workspace Reference Request
+
+The workspace reference request is sent from the client to the server to export project-wide references to dependencies. That is, the response strictly returns references in the project to symbols defined in dependencies.
```

Comment:
this may be clarified later on, but what does this specifically mean for vendored in code? I assume we want to export those. Does the language server need to be able to understand vendoring tools to decide which commits/versions the code is, or is that responsibility delegated to the build server (which understands this sort of stuff better).